storage/src/journal/variable.rs

```diff
@@ -221,6 +222,208 @@ impl<E: Storage + Metrics, V: Codec> Journal<E, V> {
         })
     }
 
+    /// Initialize a Variable journal for use in state sync.
+    ///
+    /// The bounds are operation locations (not section numbers). This function prepares the
+    /// on-disk journal so that subsequent appends go to the correct physical location for the
+    /// requested range.
+    ///
+    /// Behavior by existing on-disk state:
+    /// - Fresh (no data): returns an empty journal.
+    /// - Stale (all data strictly before `lower_bound`): destroys existing data and returns an
+    ///   empty journal.
+    /// - Overlap within [`lower_bound`, `upper_bound`]:
+    ///   - Prunes sections strictly below `lower_bound / items_per_section` (section-aligned).
+    ///   - Removes any sections strictly greater than `upper_bound / items_per_section`.
+    ///   - Truncates the final retained section so that no operation with location greater
+    ///     than `upper_bound` remains.
+    ///
+    /// Note that lower-bound pruning is section-aligned. This means the first retained section may
+    /// still contain operations whose locations are < `lower_bound`. Callers should ignore these.
+    ///
+    /// # Arguments
+    /// - `context`: storage context
+    /// - `cfg`: journal configuration
+    /// - `lower_bound`: first operation location to retain (inclusive)
+    /// - `upper_bound`: last operation location to retain (inclusive)
+    /// - `items_per_section`: number of operations per section
+    ///
+    /// # Returns
+    /// A journal whose sections satisfy:
+    /// - No section index < `lower_bound / items_per_section` exists.
+    /// - No section index > `upper_bound / items_per_section` exists.
+    /// - The last retained section is truncated so that its last item’s location is `<= upper_bound`.
+    pub(crate) async fn init_sync(
```

Comment:
I know we did this already with fixed journal, but wondering if this sync-client-specific code should be outside the core journal module. It seems more application specific?

Reply:
Fair point.

I moved the fixed journal's `init_sync` and helpers to `adb/any/fixed/sync.rs` and made them functions rather than methods.

For the variable journal, I moved them to `adb/any/variable/sync.rs`. I'm not sure this is the right place because:
1. Sync isn't yet implemented for `any::variable::Any` yet
2. These functions are used by `any::Immutable`

but I'm not immediately sure where a better place to put this code would be.

More generally, I'd like to hear others' thoughts on how we should organize the sync-related code. As it stands, there are `sync` modules for `any::fixed`, `any::variable`, `any::immutable`. An alternative would be to consolidate all sync-related implementations for these types in one place/module.